python/src/dataset.rs

```diff
@@ -941,7 +941,7 @@ impl Dataset {
             .block_on(None, operation.execute())?
             .map_err(|err| PyIOError::new_err(err.to_string()))?;
 
-        self.ds = new_self;
+        (self.ds, _) = new_self;
```

Comment:
It would be nice to also expose this in Python. But if you want to leave that as TODO, that's fine.

Reply:
added. please review. Thank you!